core/block_configuration.go

```diff
@@ -0,0 +1,7 @@
+package core
+
+type BlockConfiguration struct {
```

Comment:
I love this.
You could adjust each config to be something like this.
```
type AppConfiguration struct {
  BlockConfiguration
  ...
}
```

Reply:
I tried but it stopped parsing the yaml correctly. Instead of diving down another rabbit hole, I just made it work and left some work for my future self. :)

Reply:
Haha. Sounds good, keep plowing.